src/query/executor.rs

```diff
@@ -98,14 +104,43 @@ async fn execute_test_multiply<F: Field>(
     results
 }
 
-async fn execute_ipa<F: Field, B: BitArray>(
+async fn execute_ipa<F: Field, MK: BitArray, BK: BitArray>(
     ctx: SemiHonestContext<'_, F>,
     query_config: IPAQueryConfig,
     mut input: AlignedByteArrStream,
-) -> Vec<AggregateCreditOutputRow<F>> {
+) -> Vec<MCAggregateCreditOutputRow<F, BK>>
+where
+    IPAInputRow<F, MK, BK>: Serializable,
```

Comment:
here as well

Reply:
same advice - lets assume IPAInputRow is DST